drivers/i3c/i3c_stm32.c

```diff
@@ -473,15 +473,17 @@ static int i3c_stm32_calc_scll_od_sclh_i2c(const struct device *dev, uint32_t i2
 						1000000000ull) -
 				   1;
 			*sclh_i2c = DIV_ROUND_UP(i3c_clock, i2c_bus_freq) - *scll_od - 2;
+			if (*sclh_i2c <
+				DIV_ROUND_UP(
+					STM32_I3C_SCLH_I2C_MIN_FM_NS * i3c_clock, 1000000000ull
+				) - 1) {
```

Comment:
```suggestion
			if (*sclh_i2c < 
			      (DIV_ROUND_UP(STM32_I3C_SCLH_I2C_MIN_FM_NS * i3c_clock,
                               1000000000ull) - 1)
			   ) {
```
(Note the additional `()` by the way)

or make `STM32_I3C_SCLH_I2C_MIN_FM_NS * i3c_clock` as an intermediate value

Reply:
Not resolved